client/src/reducers/controls.js

```diff
@@ -140,30 +222,93 @@ const Controls = (
         }
       };
     }
-    case "expression load error":
-    case "initial data load error": {
+    case "request differential expression success": {
+      const { world } = state;
+      const _diffexpGenes = [];
+      action.data.forEach(d => {
+        _diffexpGenes.push(world.varAnnotations[d[0]].name);
+      });
       return {
         ...state,
-        loading: false,
-        error: action.error
+        diffexpGenes: _diffexpGenes
       };
     }
+    case "clear differential expression":
+      const { world, universe, crossfilter, dimensionMap } = state;
+      const _dimensionMap = dimensionMap;
+      let universeVarDataCache = universe.varDataCache;
+      let worldVarDataCache = world.varDataCache;
 
-    /*******************************
-             User Events
-     *******************************/
-    case "parallel coordinates axes have been drawn": {
+      _.forEach(action.diffExp, values => {
+        const name = world.varAnnotations[values[0]].name;
+        // clean up crossfilter dimensions
+        const filterID = dimensionMap[name];
+        const dimension = crossfilter.filters.find(d => {
```

Comment:
this will destroy the dimension for any user defined gene

(same name space collision noted above)